core/consensus/babe/src/lib.rs

```diff
@@ -773,11 +774,15 @@ fn claim_slot(
 	genesis_hash: &[u8],
 	epoch: u64,
 	authorities: &[AuthorityId],
-	key: &sr25519::Pair,
+	session_store: &SessionStore<AuthorityPair>,
 	threshold: u64,
-) -> Option<((VRFInOut, VRFProof, VRFProofBatchable), usize)> {
-	let public = &key.public();
-	let index = authorities.iter().position(|s| s == public)?;
+) -> Option<((VRFInOut, VRFProof, VRFProofBatchable), usize, AuthorityPair)> {
+	let (index, key) = authorities.iter()
+		.enumerate()
+		.filter_map(|(index, public)| {
+			session_store.get_key(public).map(|key| (index, key))
+		})
+		.nth(0)?;
```

Comment:
```suggestion
		.next()?;
```

Reply:
this also doesn't ensure that there is only 1 key owned by us which can author at this slot.

Reply:
the aura paper states that there is only one elected leader for any slot. so either we are the leader or not. I don't understand why this doesn't work. I've not looked at the babe paper yet...

Reply:
(we discussed offline, it's because BABE may have multiple leaders for 1 slot)